### PR TITLE
add buying outright function

### PR DESCRIPTION
**Pull request by Nelson2809**
add buying outright function
notice! mast add a table into thie topbrand database

### CODE REVIEW

File: Install [hunk 1]
```diff
@@ -28,6 +28,9 @@ ALTER TABLE `mtb_user` CHANGE `pay_points` `pay_points` INT NOT NULL DEFAULT '0'
 ALTER TABLE `mtb_order` CHANGE `give_point` `give_point` INT NOT NULL DEFAULT '0'
 ALTER TABLE `mtb_user_pointslog` CHANGE `pay_points` `pay_points` INT NOT NULL DEFAULT '0'
 
+e .直接购买
+copy the table called cart to a new table called gopay_cart
```
Comment (fhzhen) [inline Install]: Add *_ as prefix on cart & gopay_cart

File: appliaction/Controller/Goods/CartController.class.php [hunk 2]
```diff
@@ -18,7 +18,8 @@ public function _initialize() {
         $this->cart_db = model('Cart');
         $this->goods_db = model('Goods');
 		libfile('Cart');
-		$this->Cart = new Cart();
+        //changed by Nelson 2015/11/4 add buying outright function (check pay method whether add_cart or gopay)
```
Comment (fhzhen) [inline appliaction/Controller/Goods/CartController.class.php]: Please modify the comment as "Add checkout function (choice the pay method as add_cart or gopay) -- by Nelson 2015/11/4

File: appliaction/Library/Cart.class.php [hunk 3]
```diff
@@ -45,12 +51,12 @@ public function add($goods_id, $product_id = 0, $num = 1, $uid = 0) {
...
-            $item = $this->cart_db->where($sqlmap)->find();
-            if ($item) {
-                $this->cart_db->where($sqlmap)->setInc('num', $num);
-            } else {
-               $this->cart_db->update($data);
-            }
+                $item = $this->cart_db->where($sqlmap)->find();
+                if ($item) {
+                    $this->cart_db->where($sqlmap)->setInc('num', $num);
+                } else {
```
Comment (fhzhen) [inline appliaction/Library/Cart.class.php]: Format change?

Comment (Nelson2809) [inline appliaction/Library/Cart.class.php]: yes

File: template/default/statics/js/cart.js [hunk 4]
```diff
@@ -136,6 +136,16 @@ var hd_cart = (function() {
 				}
 			});
 		},
+		//changed by Nelson 2015/11/4 add buying outright function (clear the temporary database function)
+		clearBuy:function(){
+			$.getJSON('?m=goods&c=cart&a=clear&method=pay', function(ret){
+				if(ret.status == 1) {
+				} else {
```
Comment (fhzhen) [inline template/default/statics/js/cart.js]: Plesae Format these codes

**Pull request closed without merging**